### PR TITLE
Send Target.detachedFromTarget event on detach

### LINKED ISSUE #1819: Page unresponsive after Target.detachFromTarget — session_id set to null, no messages routed
After attaching to a target with `Target.attachToTarget` and then detaching with `Target.detachFromTarget`, the page stops responding to any CDP commands. No messages get a response regardless of whether you include a `sessionId` or not.

Tested with raw WebSocket (no Puppeteer), connecting to `lightpanda serve`:

```javascript
// 1. Create context + target + attach — all works fine
Target.createBrowserContext -> { browserContextId: "BID-1" }
Target.createTarget -> { targetId: "FID-0000000001" }
Target.attachToTarget -> { sessionId: "SID-1" }

// 2. Navigate + evaluate with sessionId — works
Page.navigate { url: "https://example.com", sessionId: "SID-1" } -> ok
Runtime.evaluate { expression: "document.title", sessionId: "SID-1" } -> "Example Domain"

// 3. Evaluate WITHOUT sessionId while attached — returns a result,
//    but response is tagged with sessionId: "SID-1" 
Runtime.evaluate { expression: "1+1" } -> { value: 2, sessionId: "SID-1" }

// 4. Detach
Target.detachFromTarget { sessionId: "SID-1" } -> ok

// 5. Now nothing works:
Runtime.evaluate { expression: "document.title" }  // no sessionId -> NO RESPONSE (hangs)
Runtime.evaluate { expression: "document.title", sessionId: "SID-1" } -> "Unknown sessionId"
```

So after detach you're stuck — no `sessionId` means no response, and the old `sessionId` is rejected.

Looking at the code, `detachFromTarget` in `src/cdp/domains/target.zig:396` sets `bc.session_id = null`:

```zig
fn detachFromTarget(cmd: anytype) !void {
    if (cmd.browser_context) |bc| {
        bc.session_id = null;
    }
    return cmd.sendResult(null, .{});
}
```

And then in `cdp.zig`, `isValidSessionId` rejects messages because the stored `session_id` is now `null`, and messages without a `sessionId` field seem to not get routed to the browser context at all once a session has been used.

One thing that might help: stash the previous `session_id` (or lack of one) before attach, and restore it on detach, so the connection goes back to its pre-attach state.

This also affects Puppeteer's `page.createCDPSession()` — the CDP session itself works, but the page object becomes unusable after `session.detach()` because Puppeteer routes page commands through the sessionless path.

Worth noting this is improved from v0.2.5 where `createCDPSession` would crash the WebSocket connection entirely. On nightly the connection stays alive, the page just can't receive commands anymore.

Tested on nightly (d3dad772), macOS aarch64.

### PR DESCRIPTION
**Pull request by mvanhorn**
## Summary

- `detachFromTarget` and `setAutoAttach(autoAttach: false)` both null `bc.session_id` without notifying the client. Per the [CDP spec](https://chromedevtools.github.io/devtools-protocol/tot/Target/#event-detachedFromTarget), detaching a session must fire a `Target.detachedFromTarget` event so the driver stops sending messages on the stale session ID.
- Both code paths had TODO comments calling out this exact gap: `// TODO should we send a Target.detachedFromTarget event?`
- This captures the session_id before nulling it and fires the event in both paths.

## Changes

- `detachFromTarget`: emit `Target.detachedFromTarget` with the session ID before nulling it
- `setAutoAttach(false)`: same treatment
- New test: `detachFromTarget` emits the event after attach
- New test: `detachFromTarget` without a session does not emit an event
- New test: `setAutoAttach(false)` emits the event after attach

## Test plan

- [x] All 368 existing tests pass (`zig build test`)
- [x] New tests cover event emission in both detach paths
- [x] Edge case: detach with no active session does not crash or emit

Fixes #1819

This contribution was developed with AI assistance (Claude Code + Codex).

### CODE REVIEW

**Comment by krichprollsch** [comment]
Thanks for your contribution :pray:
